homeassistant/components/august/__init__.py

```diff
@@ -136,23 +138,27 @@ def setup_august(hass, config, api, authenticator):
         if DOMAIN in _CONFIGURING:
             hass.components.configurator.request_done(_CONFIGURING.pop(DOMAIN))
 
-        hass.data[DATA_AUGUST] = AugustData(hass, api, authentication, authenticator)
+        hass.data[DATA_AUGUST] = AugustData(
+            hass, api, authentication, authenticator, token_refresh_lock
+        )
 
         for component in AUGUST_COMPONENTS:
-            discovery.load_platform(hass, component, DOMAIN, {}, config)
+            hass.async_create_task(
```

Comment:
We can't call an async function inside a sync function.

Reply:
You should use `hass.add_job`

Reply:
But in that case you might as well go back to calling `discovery.load_platform` like it was before.

Reply:
Also, calling a sync function here will not impact the fact that the platform is using an async_setup_platform function.

Reply:
Will revert